cmd/tetra/common/flags.go

```diff
@@ -26,3 +35,44 @@ var (
 	Timeout       time.Duration
 	Retries       int
 )
+
+func readActiveServerAddressFromFile(fname string) (string, error) {
+	f, err := os.Open(fname)
+	if err != nil {
+		return "", fmt.Errorf("failed to open init info file: %w", err)
+	}
+	defer f.Close()
+
+	var info struct {
+		ServerAddr string `json:"server_address"`
+	}
+	if err := json.NewDecoder(f).Decode(&info); err != nil {
+		return "", fmt.Errorf("failed to decode init info file: %w", err)
+	}
+
+	return info.ServerAddr, nil
+}
+
+// ResolveServerAdress returns the server address given by the user from the
+// command line flag, if not set, try to read from tetragon-info.json, and, if
+// the file doesn't exist, returns the default value.
+func ResolveServerAddress() (string, error) {
+	if ServerAddress == "" {
+		sa, err := readActiveServerAddressFromFile(defaults.InitInfoFile)
+
+		if err != nil {
+			logger.GetLogger().WithError(err).WithField(
+				"defaultServerAddress", defaultServerAddress,
+			).Debug("failed to resolve server address reading init info file, using default value")
+			return defaultServerAddress, nil
+		}
+
+		logger.GetLogger().WithFields(logrus.Fields{
+			"InitInfoFile":  defaults.InitInfoFile,
+			"ServerAddress": sa,
+		}).Debug("resolved server address using info file")
+		return sa, nil
+	}
+
+	return ServerAddress, nil
+}
```

Comment:
We're always returning nil here. Perhaps we could consider removing the error return in the type signature?

Reply:
indeed thanks